website/scripts/write-heading-ids.js

```diff
@@ -0,0 +1,118 @@
+/** Write custom heading ids for all files in the docs directory */
+const fs = require('fs/promises');
```

Comment:
It would be good to document this more extensively so that more of us could help maintain this.  

Apart from documenting the individual functions, What is the purpose, i.e. why do we need custom heading ids, etc.

Also would be cool if we could run this with ts-node so we could type it properly.